docs/reference/components/sources/aws_s3.cue

```diff
@@ -0,0 +1,225 @@
+						}
+					}
+				}
+			}
+		}
+	}
+
+	output: logs: object: {
+		description: "A line from an S3 object."
+		fields: {
+			message: {
+				description: "A line from the S3 object."
+				required:    true
+				type: string: examples: ["53.126.150.246 - - [01/Oct/2020:11:25:58 -0400] \"GET /disintermediate HTTP/2.0\" 401 20308"]
+			}
+			timestamp: fields._current_timestamp & {
+				description: "The Last-Modified time of the object. Defaults the current timestamp if this information is missing."
+			}
+			bucket: {
+				description: "The bucket of the object the line came from."
+				required:    true
+				type: string: examples: ["my-bucket"]
+			}
+			object: {
+				description: "The object the line came from."
+				required:    true
+				type: string: examples: ["AWSLogs/111111111111/vpcflowlogs/us-east-1/2020/10/26/111111111111_vpcflowlogs_us-east-1_fl-0c5605d9f1baf680d_20201026T1950Z_b1ea4a7a.log.gz"]
+			}
+			region: {
+				description: "The AWS region bucket is in."
+				required:    true
+				type: string: examples: ["us-east-1"]
+			}
+		}
+	}
+
+	how_it_works: {
+		events: {
+			title: "Handling events from the `aws_s3` source"
+			body: """
+				This source behaves very similarly to the `file` source in that
+				it will output one event per line (unless the `multiline`
+				configuration option is used).
+
+				You will commonly want to use [transforms][urls.vector_transforms] to
+				parse the data.  For example, to parse VPC flow logs sent to S3 you can
+				chain the `tokenizer` transform:
+
+				```toml
+				[transforms.flow_logs]
+					type = "tokenizer" # required
+					inputs = ["s3"]
+					field_names = ["version", "account_id", "interface_id", "srcaddr", "dstaddr", "srcport", "dstport", "protocol", "packets", "bytes", "start", "end", "action", "log_status"]
+
+					types.srcport = "int"
+					types.dstport = "int"
+					types.packets = "int"
+					types.bytes = "int"
+					types.start = "timestamp|%s"
+					types.end = "timestamp|%s"
+				```
+
+				To parse AWS load balancer logs, the `regex_parser` transform can be used:
+
+				```toml
+				[transforms.elasticloadbalancing_fields_parsed]
+					type = "regex_parser"
+					inputs = ["s3"]
+					regex = '^(?P<type>[\\w]+) (?P<timestamp>[\\w:.-]+) (?P<elb>[^\\s]+) (?P<client_host>[\\d.:-]+) (?P<target_host>[\\d.:-]+) (?P<request_processing_time>[\\d.-]+) (?P<target_processing_time>[\\d.-]+) (?P<response_processing_time>[\\d.-]+) (?P<elb_status_code>[\\d-]+) (?P<target_status_code>[\\d-]+) (?P<received_bytes>[\\d-]+) (?P<sent_bytes>[\\d-]+) "(?P<request_method>[\\w-]+) (?P<request_url>[^\\s]+) (?P<request_protocol>[^"\\s]+)" "(?P<user_agent>[^"]+)" (?P<ssl_cipher>[^\\s]+) (?P<ssl_protocol>[^\\s]+) (?P<target_group_arn>[\\w.:/-]+) "(?P<trace_id>[^\\s"]+)" "(?P<domain_name>[^\\s"]+)" "(?P<chosen_cert_arn>[\\w:./-]+)" (?P<matched_rule_priority>[\\d-]+) (?P<request_creation_time>[\\w.:-]+) "(?P<actions_executed>[\\w,-]+)" "(?P<redirect_url>[^"]+)" "(?P<error_reason>[^"]+)"'
```

Comment:
nit: might be worthwhile to split this regex over multiple lines for readability (and enable the flag that makes the regex ignore whitespace and comments).